lib/connection.ts

```diff
@@ -49,8 +49,40 @@ export interface ConnectionOptions extends RheaConnectionOptions {
    * timeout occurs. Default: `60 seconds`.
    */
   operationTimeoutInSeconds?: number;
+
+  /**
+   * @property {Object} [webSocketOptions] - Options that include a web socket constructor along
+   * with arguments to be passed to the function returned by rhea.websocket_connect()
+   * This is required when the connection needs to use web sockets but is being created without
+   * creating a container first. If a container is already available, use `websocket_connect` on it
+   * directly instead.
+   */
+  webSocketOptions?: {
+    /**
+     * @property {any} [webSocket] - The WebSocket constructor used to create an AMQP
+     * connection over a WebSocket.
+     */
+    webSocket: any;
```

Comment:
You can type webSocket [more precisely here](https://github.com/Azure/amqp-common-js/pull/29/files#diff-ad0ce2742650d03b78af445d57e30269R6), which I think is fairly critical because there are websocket implementations that expose differently-shaped constructors (e.g. `ws` for Node), and you can't just pass `ws` directly as `webSocket` because Rhea will throw attempting to construct it.

Reply:
I did exactly that in my first attempt, but then was doubtful if rhea supported other types as well and the types we are talking about is more specific to our needs.

Reply:
FWIW I constructed those types by examining every API used by Rhea so it should document the minimum-required interface to work properly.

Reply:
On second thoughts, these types are best placed in the `rhea` library than `rhea-promise`.
Towards that, I have created the issue https://github.com/amqp/rhea/issues/209

Since this is not blocker for our browserification attempts, we can go ahead with this PR